node-test-loader.js

```diff
@@ -0,0 +1,36 @@
+import { readFileSync } from 'fs';
+import { fileURLToPath } from 'url';
+
+// input object for testing
+export function globalPreload() {
+  return `globalThis.input = {
+    set works (val) {
+      if (val)
+        console.log('SUCCESS');
+    }
+  };`;
+}
+
+export async function resolve (specifier, context, _defaultResolve) {
+  const url = new URL(specifier, context.parentURL);
+
+  // extension adding
+  if (!url.pathname.endsWith('.js') && !url.pathname.endsWith('.mjs') && !url.pathname.endsWith('.cjs')) {
+    url.pathname += '.js';
+  }
```

Comment:
The extension adding here should probably be extended to do actual statting with variations too.